core/src/cache.rs

```diff
@@ -1545,6 +1598,43 @@ fn normalize_abs_path(path: &Path) -> PathBuf {
     ret
 }
 
+/// Normalize a relative path, removing mid-path `..`s.
+///
+/// Like [`normalize_abs_path`], this works only on the path itself
+/// (i.e. not the filesystem) and does not follow symlinks.
+pub fn normalize_rel_path(path: &Path) -> PathBuf {
```

Comment:
I needed this utility function in the upcoming `nickel_lang_package` crate. Putting it next to `normalize_abs_path` seemed reasonable, but it doesn't actually need to be in this crate.